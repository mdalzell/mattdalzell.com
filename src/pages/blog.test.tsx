import React from 'react'
import renderer from 'react-test-renderer'

// @ts-ignore
import * as gatsby from '../../__mocks__/gatsby'

import BlogPage from './blog'

interface IStaticQuery {
  render: (object: {
    site: { siteMetadata: { title: string } }
  }) => React.ReactNode
}

beforeEach(() => {
  gatsby.StaticQuery.mockImplementationOnce(({ render }: IStaticQuery) =>
    render({
      site: {
        siteMetadata: {
          title: `Matt Dalzell - Ace Programmer`,
        },
      },
    })
  )
})

describe('BlogPage', () => {
  const mockProps = {
    data: {
      allMarkdownRemark: {
        edges: [
          {
            node: {
              fields: { slug: 'test-1' },
              frontmatter: {
                date: new Date('5/1/2019').toDateString(),
                tags: ['red', 'blue'],
                title: 'Test #1',
              },
              id: 1,
            },
          },
          {
            node: {
              fields: { slug: '/test-2' },
              frontmatter: {
                date: new Date('4/1/2019').toDateString(),
                tags: ['red'],
                title: 'Test #2',
              },
              id: 2,
            },
          },
        ],
        group: [
          { fieldValue: 'red', totalCount: 2 },
          { fieldValue: 'blue', totalCount: 1 },
        ],
      },
    },
    location: {
      search: '',
    },
  }
  it('should match snapshot for default view', () => {
    const component = renderer.create(<BlogPage {...mockProps} />).toJSON()

    expect(component).toMatchSnapshot()
  })
})
