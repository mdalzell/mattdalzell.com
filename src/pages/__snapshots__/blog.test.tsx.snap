// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPage should match snapshot for default view 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-3 hidden-sm"
      >
        <ul
          className="nav nav-list"
        >
          <li>
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/blog"
            >
              Blog
            </a>
          </li>
          <li>
            <a
              rel="noopener noreferrer"
              target="_blank"
            >
              <span>
                <span
                  className="glyphicon glyphicon-list-alt"
                />
                Resume
              </span>
            </a>
          </li>
          <li>
            <a
              href="https://github.com/mdalzell"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span>
                <span
                  className="glyphicon glyphicon-wrench"
                />
                GitHub
              </span>
            </a>
          </li>
          <li>
            <a
              href="https://stackoverflow.com/story/mattdalzell"
              rel="noopener noreferrer"
              target="_blank"
            >
              <span>
                <span
                  className="glyphicon glyphicon-console"
                />
                Stack Overflow
              </span>
            </a>
          </li>
          <li>
            <a
              href="mailto:contact@mattdalzell.com"
            >
              Email
            </a>
          </li>
        </ul>
      </div>
      <div
        className="col-sm-9"
      >
        <div>
          <h1>
            Blog
          </h1>
          <h3>
            Recent Posts
          </h3>
          <ul
            className="no-list-style"
          >
            <li>
              <a
                href="/blogtest-1"
              >
                Test #1 - Wed May 01 2019
              </a>
            </li>
            <li>
              <a
                href="/blog/test-2"
              >
                Test #2 - Mon Apr 01 2019
              </a>
            </li>
          </ul>
          <h3>
            Tags
          </h3>
          <p>
            <a
              className="tag"
              href="/blog?tag=red"
            >
              #red
            </a>
            <a
              className="tag"
              href="/blog?tag=blue"
            >
              #blue
            </a>
          </p>
          <a
            href="/"
          >
            Return Home
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <ul
      className="nav nav-list"
    >
      <li>
        <a
          href="/"
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/blog"
        >
          Blog
        </a>
      </li>
      <li>
        <a
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            <span
              className="glyphicon glyphicon-list-alt"
            />
            Resume
          </span>
        </a>
      </li>
      <li>
        <a
          href="https://github.com/mdalzell"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            <span
              className="glyphicon glyphicon-wrench"
            />
            GitHub
          </span>
        </a>
      </li>
      <li>
        <a
          href="https://stackoverflow.com/story/mattdalzell"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span>
            <span
              className="glyphicon glyphicon-console"
            />
            Stack Overflow
          </span>
        </a>
      </li>
      <li>
        <a
          href="mailto:contact@mattdalzell.com"
        >
          Email
        </a>
      </li>
    </ul>
    <div>
      Â© 
      2019
       Matt Dalzell
    </div>
  </div>
</div>
`;
